### PR TITLE
Verifying the hash of downloaded Bazel binary

### PR DESCRIPTION
**Pull request by linzhp**
In some areas with unstable or slow internet, the downloaded Bazel may be corrupted. We should verify the hash of the downloaded binary, delete the binary and prompt users to try again if it is corrupted. For backward compatibility for existing mirrors that don't have sha256 file, we will skip the verification if the ".sha256" file doesn't exist.

### CODE REVIEW

**Comment by linzhp** [comment]
@philwo Would you mind also taking a look at this?

**Comment by linzhp** [comment]
@fweikert Would you mind taking a look at this too?

File: bazelisk.py [hunk 1]
```diff
@@ -268,30 +272,59 @@ def download_bazel_into_directory(version, is_commit, directory):
...
+    actual_hash = sha256_hash.hexdigest()
+    if actual_hash != expected_hash:
+        os.remove(destination_path)
+        os.remove(sha256_path)
+        print(
+            "The downloaded Bazel binary is corrupted. Expected SHA-256 {}, got {}. Please try again.".format(
+                expected_hash, actual_hash
+            )
+        )
+        # Exiting with a special exit code not used by Bazel, so the calling process may retry based on that.
```
Comment (fweikert) [inline bazelisk.py:305]: Current Bazel versions use 45: https://docs.bazel.build/versions/5.0.0/guide.html#what-exit-code-will-i-get

**Review by fweikert — CHANGES_REQUESTED** [review]

File: bazelisk.py (hunk 1 above)
Comment (linzhp) [reply bazelisk.py:330]: Changed it to 22

**Review by fweikert — APPROVED** [review]

**Pull request merged**